### ISSUE TITLE
Allow boolean arrays for trajectory indexing

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#725]
It would be great if code like this would work

```python
import MDAnalysis as mda
import numpy as np
from MDAnalysisTests.datafiles import PSF, DCD

u  = mda.Universe(PSF, DCD)
range_ = np.arange(len(u.trajectory))
mask = (range_ % 2) == 0

for ts in u.trajectory[mask]:
    continue
```

That would make it easy to select subsets of a trajectory with a mask. Something I do often after a state assignment.

**Comment by richardjgowers** [comment]
Yeah should be easily possible [we've got it for AtomGroups](https://github.com/MDAnalysis/mdanalysis/blob/78992cffd40682ffa644c486ccbc638232497ae4/package/MDAnalysis/core/AtomGroup.py#L1033), it should be fairly easy to move this into base.Reader

**Comment by orbeckst** [comment]
I'd like it, too.
One more feature that makes trajectories work more seamlessly in the existing python environment and reduces *user interface shear* (where the user is made to believe that something works like all the other things they've been working with but then crucial elements suddenly behave differently --- similar to the concept of  *metaphor shear* introduced in Neal Stephenson's [In the beginning was the command line](http://www.cryptonomicon.com/beginning.html) (search the [text](http://cristal.inria.fr/~weis/info/commandline.html) or [PDF](http://becksteinlab.physics.asu.edu/files/NealStephenson_Commandline.pdf))).

**Comment by richardjgowers** [comment]
Closed in #730
